spec/system/managing_topics_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe 'managing topics' do
+  before do
+    FactoryBot.create(:topic, name: 'What is your favourite colour')
+  end
+
+  scenario 'when viewing' do
+    visit('/topics')
+    expect(page).to have_content('Topics')
+    expect(page).to have_content('What is your favourite colour')
+  end
+
+  scenario 'can navigate to new topic page' do
+    visit('/topics')
+    expect(page).to have_link('New Topic',href: new_topic_path)
+  end
+
+  scenario 'when creating a topic' do
+    visit new_topic_path
+    fill_in :name, with: 'tell us a story'
+    click_button 'Create'
+    expect(current_path).to eq topics_path
+    expect(page).to have_content 'tell us a story'
+
+    visit '/topics/new'
```

Comment:
This whole system spec is a bit muddled from a user perspective. It is supposed to represent a user trying to create a topic. Why is the user going back to the new topic page, starting a new topic creation and abandoning it after getting a validation warning. 

An example for why this doesn't work as a test at the moment is that you haven't verified that it is possible to get past the validation message without a page refresh.

Here are the steps in an order that makes sense from a user perspective
* visit the topics index
* click the _New topic_ link
* Hit _Create Topic_ button
* Whoops! Validation messages say I've done something wrong
* Enter the topic name
* Hit _Create Topic_ button again
* Redirected to Topics#index page
* Check the new topic is there
* Ahhh relief.

Your test should make sure that this user operation/journey works.

Reply:
Assuming all the tests are currently passing, the way you've laid your test out may be hiding the error I talk about in the controller create action (see other comment).

Reply:
It does seem my programming style is 'arse about face' sometimes :/